File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/webforms/busobjs/cs/BusObj.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System.Reflection;

[assembly: AssemblyTitle("")]
[assembly: AssemblyDescription("A QuickStart Tutorial Assembly")]
[assembly: AssemblyConfiguration("")]
[assembly: AssemblyCompany("Microsoft Corporation")]
[assembly: AssemblyProduct("Microsoft QuickStart Tutorials")]
[assembly: AssemblyCopyright(" Microsoft Corporation.  All rights reserved.")]
[assembly: AssemblyTrademark("")]
[assembly: AssemblyCulture("")]
[assembly: AssemblyVersion("1.1.*")]

namespace BusinessLayer
{
using System;
using System.Data;
using System.Data.SqlClient;
using DataLayer;

  public class BusObj
  {
    private DataObj data;

    public BusObj()
    {
      data = new DataObj("server=(local)\\NetSDK;database=grocertogo;Integrated Security=SSPI");
    }

    public DataView GetCategories()
    {
       return data.GetCategories();
    }

    public DataView GetProductsForCategory(String category, int customerid)
    {
       DataView view = data.GetProductsForCategory(category);

       double discount = 0;
       if ((customerid >= 25)&&(customerid < 50))
       {
         discount = .50;
       }
       else if ((customerid >= 50)&&(customerid < 75))
       {
         discount = 1.00;
       }
       else if ((customerid >= 75)&&(customerid < 100))
       {
         discount = 1.50;
       }

       for (int i=0; i<view.Count; i++)
       {
         view[i]["UnitPrice"] = Double.Parse(view[i]["UnitPrice"].ToString()) - discount;
       }

       return view;
    }
  }

}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/aspplus/samples/webforms/busobjs/vb/BusObj.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports DataLayer
Imports System.Reflection

< _
    Assembly : AssemblyTitle(""), _
    Assembly : AssemblyDescription("A QuickStart Tutorial Assembly"), _
    Assembly : AssemblyConfiguration(""), _
    Assembly : AssemblyCompany("Microsoft Corporation"), _
    Assembly : AssemblyProduct("Microsoft QuickStart Tutorials"), _
    Assembly : AssemblyCopyright(" Microsoft Corporation.  All rights reserved."), _
    Assembly : AssemblyTrademark(""), _
    Assembly : AssemblyCulture(""), _
    Assembly : AssemblyVersion("1.1.*"), _
    Assembly : AssemblyDelaySign(false), _
    Assembly : AssemblyKeyFile(""), _
    Assembly : AssemblyKeyName("") _
>
Namespace BusinessLayer

  Public Class BusObjVB

    Private Data As DataObjVB

    Public Sub New()

      MyBase.New()
      Data = New DataObjVB("server=(local)\NetSDK;database=grocertogo;Integrated Security=SSPI")
    End Sub

    Public Function GetCategories() As DataView

       Return Data.GetCategories()
    End Function

    Public Function GetProductsForCategory(Category As String, CustomerId As Integer) As DataView

       Dim View As DataView
       View = Data.GetProductsForCategory(Category)

       Dim Discount As Double = 0
       If (CustomerId >= 25) And (CustomerId < 50)
         Discount = .50
       Else If (CustomerId >= 50) And (CustomerId < 75)
         Discount = 1.00
       Else If (CustomerId >= 75) And (CustomerId < 100)
         Discount = 1.50
       End If

       Dim I As Integer
       For I = 0 To View.Count - 1
         View(I)("UnitPrice") = CDbl(View(I)("UnitPrice").ToString()) - Discount
       Next

       Return View
    End Function

  End Class

End Namespace